packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -138,15 +191,28 @@ void main() {
 }
 
 class CrashingFlutterCommand extends FlutterCommand {
+  CrashingFlutterCommand({this.direct = true});
+
+  final bool direct;
+
   @override
   String get description => null;
 
   @override
-  String get name => 'test';
+  String get name => 'crash';
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    throw 'an exception % --'; // Test URL encoding.
+    const String error = 'an exception % --'; // Test URL encoding.
+    if (direct) {
+      throw error;
+    }
+
+    Timer.run(() => throw error);
+
+    // Give the Timer time to fire.
+    await Future<void>.delayed(const Duration(milliseconds: 100));
```

Comment:
It might work in this case, but it's generally not a good idea to have explicit timers like this in tests. There are two options. The first option is to use a `FakeAsync` block. I think that might be overkill here. Instead, it looks like you might be able to pass around some `Future`s to await and some `Completer`s to complete to achieve the execution order that you're trying to exercise.

Reply:
I've replaced this one with a `Completer` to wait on.  I didn't think that particular case would be controversial since it involves a 0-delay timer and a non-zero delay timer, and I expect that timer callbacks would fire in order.

I really don't like the `Timer` in `SlowCrashReporter`.  The point of that is to make `runner.run`'s failure path take longer than the success path if they're racing.  I don't think there's anything along the success path that I can wait on; I can't add a `Completer` to the test-implementation of `exit()` since the fix is to *not* call `exit()` in this situation.

I could add an explicit, global, `@visibleForTesting`-only `Completer` to `runner.dart` that will be completed when we continue down the success path.  That also feels gross, but I suppose it's better than the hard-coded delay.  Is that acceptable?